test.zsh

```diff
@@ -0,0 +1,22 @@
+# https://opensource.apple.com/source/zsh/zsh-65/zsh/Doc/help/
+# https://gist.github.com/cinhtau/ea0bed82dfa8f996c3cd4149c2c31672
+
+local -a just_values
+typeset -A values_sorted
+typeset -A history_my
+typeset -A values
+# just_values=(a b c)
+history_my=([1]=d [2]=e  [3]=f [4]=e)
+print ${(kv@)history_my}
+# print "$s" ${(@)just_values}
+for key val in "${(@kv)history_my}"; do
+  # print $val
+  values[$val]=$key
+  just_values+=$val
+done
+
+for val in "${(@vu)just_values}"; do
+  values_sorted[$values[$val]]=$val 
+done
+print ${(kv@)values_sorted}
+#  print -f '%s:%s' "${(kv@)values_sorted}"
```

Comment:
This is what you should be doing:
```zsh
% local -A tmp=(a 1 b 2 c 2)
% tmp=(${(kv@Oa)tmp})
% print ${(kv)tmp}
1 a 2 b
% tmp=(${(kv@Oa)tmp})
% print ${(kv)tmp}
a 1 b 2
%
```
As you can see, swapping the keys with the values eliminates all duplicate values, since it makes the values into keys and there cannot be duplicate keys in an associative array. Then all that remains is to swap them again, so that your original values are treated as values again, not keys.

Reply:
I probably should have asked this earlier, but where are you finding your documentation for zsh? I haven't been able to find any good docs and mostly have been shooting in the dark.

Reply:
This is the official documentation: http://zsh.sourceforge.net/Doc/Release/index.html